boilerplate/app/utils/useHeader.tsx

```diff
@@ -13,10 +14,15 @@ export function useHeader(
   deps: Parameters<typeof useLayoutEffect>[1] = [],
 ) {
   const navigation = useNavigation()
+  /**
+   * We need to have multiple implementations of this hook for web and mobile.
+   * Web needs to use useEffect to avoid a rendering loop.
+   * In mobile and also to avoid a visible header jump when navigating between screens, we use
+   * `useLayoutEffect`, which will apply the settings before the screen renders.
+   */
+  const useAppropriateEffect = Platform.OS === "web" ? useEffect : useLayoutEffect
```

Comment:
nit, just to make it clear what the criteria for the effect is:
```suggestion
  const usePlatformEffect = Platform.OS === "web" ? useEffect : useLayoutEffect
```